design-proposals/migration-target.md

```diff
@@ -0,0 +1,216 @@
+With the same mechanism a k8s user can already [constrain a generic Pod so that it is restricted to run on particular node](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/).
+This proposal is aiming to expose the same capability to cluster admins, in a fully declarative way without the need to mess up the spec of the existing VM and revert at the end, when dealing with live migrations.
+
+## Motivation
+In the ideal cloud native design, the scheduler is supposed to be always able to correctly identify
+the best node to run a pod (so the target pod for the VMI after the live-migration) on.
+In the real world, we still see specific use cases where the flexibility do explicitly and directly define the target node for a live migration is a relevant nice-to-have:
+- Experienced admins are used to control where their critical workloads are move to 
+  > I as an admin, notice that a VM with guaranteed resources is having issues (I watched the cpu iowait metric). In order to resolve the performance issue and keep my user happy, I as admin want to move the VM, without interruption, to a node which is currently underutilized - and will make the user's vm perform better.
+- Workload balancing solution doesn't always work as expected
+  > I have configured my cluster with the descheduler and a load aware scheduler (trimaran), thus by default, my VMs will be regularly descheduled if utilization is not balanced, and trimaran will ensure that my VMs will be scheduled to underutilized nodes. Often this is working, however, in exceptional cases, i.e. if the load changes too quickly, or only 1 VM is suffering, and I want to avoid that all Vms on the cluster are moved, I need - for exception - a tool to move one VM, once to deal with this exceptional situation.
+- Troubleshooting a node
+- Validating a new node migrating there a specific VM
+  > I, as a battle tested admin, ordered a new node, because the old one broken, got the HW team to install it in the dc, ask the net team to wire it up, installed it myself, and included it in the cluster - tainted.
+  I drink a coffee, as this was a lot of work.
+  Now I want to move one of my own VMs over to it to do some sanity testing.
+  No other Vm on my 50 node cluster should be impacted. I just want my very own vm to move over and see if all behaves well.
+  I've seen enough issues with misbehaving hardware, misbehaving live migrations, cpu firmware issues, broken storage etc. this is my sanity test
+
+> [!IMPORTANT]
+> Directly selecting named nodes as destinations is not assumed to be a default tool for balancing workloads or all the use-cases above. It's instead just a convenient tool for exceptional situations and one-offs to ensure that and admin can quickly react to emergencies, and spikes.
```

Comment:
Are we going to document this in the user-guide for this feature?

Reply:
Yes, we are.